cirq/ops/two_qubit_noise_channels.py

```diff
@@ -0,0 +1,1139 @@
+        if args.precision is not None:
+            f = '{:.' + str(args.precision) + 'g}'
+            return 'A({},{},{},{},{},{},{},{},{},{},{},{},{},{},{},{})' \
+                .format(f, f, f, f, f, f, f, f, f, f, f, f, f, f, f, f, ).format(
+                        self._p_xi, self._p_xx, self._p_xy, self._p_xz,
+                        self._p_yi, self._p_yx, self._p_yy, self._p_yz,
+                        self._p_zi, self._p_zx, self._p_zy, self._p_zz,
+                        self._p_iz, self._p_ix, self._p_iy)
+        return 'A({!r},{!r},{!r},{!r},{!r},{!r},{!r},{!r},{!r},{!r},{!r},{!r},{!r},{!r},{!r})'.format(
+            self._p_xi, self._p_xx, self._p_xy, self._p_xz, self._p_yi,
+            self._p_yx, self._p_yy, self._p_yz, self._p_zi, self._p_zx,
+            self._p_zy, self._p_zz, self._p_iz, self._p_ix, self._p_iy)
+
+
+def two_qubit_asymmetric_depolarize(p_xi: float, p_yi: float, p_zi: float,
+                                    p_xx: float, p_yx: float, p_zx: float,
+                                    p_xy: float, p_yy: float, p_zy: float,
+                                    p_xz: float, p_yz: float, p_zz: float,
+                                    p_ix: float, p_iy: float, p_iz: float
+                                   ) -> TwoQubitAsymmetricDepolarizingChannel:
+    """
+    Returns a TwoQubitAsymmetricDepolarisingChannel with the given probabilities.
+    :param p_xi: The probability that a Pauli X on qubit 1 and no other gate occurs.
+    :param p_yi: The probability that a Pauli Y on qubit 1 and no other gate occurs.
+    :param p_zi: The probability that a Pauli Z on qubit 1 and no other gate occurs.
+    :param p_xx: The probability that a Pauli X on qubit 1 and a Pauli X on qubit 2 occurs.
+    :param p_yx: The probability that a Pauli Y on qubit 1 and a Pauli Y on qubit 2 occurs.
+    :param p_zx: The probability that a Pauli Z on qubit 1 and a Pauli Z on qubit 2 occurs.
+    :param p_xy: The probability that a Pauli X on qubit 1 and a Pauli X on qubit 2 occurs.
+    :param p_yy: The probability that a Pauli Y on qubit 1 and a Pauli Y on qubit 2 occurs.
+    :param p_zy: The probability that a Pauli Z on qubit 1 and a Pauli Z on qubit 2 occurs.
+    :param p_xz: The probability that a Pauli X on qubit 1 and a Pauli X on qubit 2 occurs.
+    :param p_yz: The probability that a Pauli Y on qubit 1 and a Pauli Y on qubit 2 occurs.
+    :param p_zz: The probability that a Pauli Z on qubit 1 and a Pauli Z on qubit 2 occurs.
+    :return: A two qubit asymmetric depolarising channel with the given probabilities.
+    """
+    return TwoQubitAsymmetricDepolarizingChannel(p_xi, p_yi, p_zi, p_xx, p_yx,
+                                                 p_zx, p_xy, p_yy, p_zy, p_xz,
+                                                 p_yz, p_zz, p_ix, p_iy, p_iz)
+
+
+@cirq.value.value_equality
+class TwoQubitDepolarizingChannel(gate_features.TwoQubitGate):
```

Comment:
Move this into a file `cirq/contrib/noise_models/two_qubit_noise_models.py`. Also add a file that tests it in that `noise_models_directory`?